uvicorn/workers.py

```diff
@@ -62,7 +63,10 @@ def init_process(self):
         super(UvicornWorker, self).init_process()
 
     def init_signals(self):
-        pass
+        # Reset signals so Gunicorn doesn't swallow subprocess return codes
+        # other signals are set up by Server.install_signal_handlers()
+        for s in self.SIGNALS:
+            signal.signal(s, signal.SIG_DFL)
```

Comment:
Yeah, that was my first intuition as well.

But then I realized that the server installs the signal handlers it supports in a way that is compatible with asyncio here on `Server.install_signal_handlers()`: https://github.com/encode/uvicorn/blob/master/uvicorn/server.py#L242

I suspect that once those signals were added to git, this Gunicorn worker subclass method was changed to just override the parent class and have just the `pass`. To force it to *not* install the default handlers, as Uvicorn has its own. But then that previous change also removed the signal resets (which are still needed).

What I understand we need is 3 things:

* Reset the Gunicorn `Arbiter` signal handlers in the `UvicornWorker`, so that those signals are not used by the worker processes. That is done here in this PR.
* Override the default Gunicorn method `init_signals()`, to skip those signal handlers that are not compatible with asyncio. As those are not compatible with asyncio. That was done by the previous `pass` here, and is now done here by the fact that the method is still overridden and the previous signal handlers are not copied.
* Install the supported signal handlers in a way compatible with asyncio. That is done in [`Server.install_signal_handlers()`](https://github.com/encode/uvicorn/blob/master/uvicorn/server.py#L242) already.